### PR TITLE
pass framedim to _base_plot

### PR DESCRIPTION
**Pull request by timothyas**
Hey @jbusecke! First off, thank you for this package!! I remember writing, rewriting, and never really having something I liked to make movies way back when I used MATLAB. But this is just super easy to use ... kudos.

This is a simple PR to pass a `framedim` other than time from the `Movie` initialization to `_base_plot`. It could also be tweaked to pass `timestep` as a dict with the dimension and index, but I figured I'd see what you think about this first.

I modified `test_Movie` to rename the "time" coordinate when framedim is something else, which seemed like a sufficient test. I was able to test everything except those that required cv2 - I had some difficulty with lower level libraries with that one so couldn't test it locally, and something is up with GitHub Actions so CI is not running on my separate branches of this fork.

### CODE REVIEW

File: xmovie/presets.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def _core_plot(ax, data, plotmethod=None, **kwargs):
     return p
 
 
-def _base_plot(ax, base_data, timestamp, plotmethod=None, **kwargs):
+def _base_plot(ax, base_data, timestamp, framedim, plotmethod=None, **kwargs):
     # need to convert time to input variable
```
Comment (jbusecke) [inline xmovie/presets.py:68]: I think that this comment was my attempt to eventually fix this 🤦‍♂️. You could probably remove it!

File: xmovie/presets.py [hunk 2]
```diff
@@ -215,12 +215,12 @@ def _add_coast(ax, style):
 
 ### Presets (should proabably put all others into a submodule)
 def basic(
-    da, fig, timestamp, plotmethod=None, plot_variable=None, subplot_kw=None, **kwargs
+    da, fig, timestamp, framedim="time", plotmethod=None, plot_variable=None, subplot_kw=None, **kwargs
```
Comment (jbusecke) [inline xmovie/presets.py:218]: I like this way of doing it. The public presets have framedim as kwarg and the private ones as an explicit arg! Nice.

**Review by jbusecke — CHANGES_REQUESTED** [review]
This looks great @timothyas! Some smaller nits as inline comments. Do you think it would be possible to add an example for the xarray test dataset, where the movie just moves through the x or y dimension while showing time vs the other dimension? I think that would help with documenting this feature better. Also don't forget to add yourself to the `whats-new.rst` It will become public once I finish the documentation.

File: xmovie/presets.py (hunk 1 above)
Comment (timothyas) [reply xmovie/presets.py]: Hey at least it was documented! Actually this confirmed for me where the change needed to happen. Either way, done!

**Comment by jbusecke** [comment]
Oh, I guess I didn't set this package up for a community contributions yet. But this seems like the time to do it. If you could do that, that would be amazing!￼

**Comment by timothyas** [comment]
No problem, whats-new is newly added ... I borrowed the style from xgcm, I figured you'd like that one ;)

**Comment by timothyas** [comment]
The CI threw some crazy error when I pushed 132d89c , saying that something in the workflow was deprecated. I updated the setup miniconda line to use conda-incubator v2 instead of goanpeca and it seems to be good now 🤷‍♂️

**Comment by timothyas** [comment]
Hey @jbusecke, just refreshing this - does this look good to you? Thanks!

**Comment by jbusecke** [comment]
So sorry to let this sit for too long! I'll try to take a better look this afternoon!

**Comment by timothyas** [comment]
No worries and no rush! I just randomly saw it was still here. Thanks

**Comment by jbusecke** [comment]
Sorry again, I have not forgotten about this (again). Just still very busy. Thanks for the patience.

**Comment by timothyas** [comment]
No worries, I hear you

**Comment by jbusecke** [comment]
This is wonderful. Many thanks for adding the example and this fix @timothyas

**Comment by timothyas** [comment]
Woohoo! No problem!

**Pull request merged**